source/contributed/caching-overview.md

```diff
@@ -0,0 +1,76 @@
+        }
+        return runExpensiveCodeResults
+    }
+
+These two examples have a drawback in that they are only run or defined when the code is first loaded. This can be bypassed by accessing the `[global](https://nodejs.org/api/globals.html#globals_global)` object, which is a special object in Node that is accessible everywhere.
+
+    function runExpensiveCode() {
+        if(!global.runExpensiveCodeResults) {
+            global.runExpensiveCodeResults = someExpensiveCode()
+        }
+        return global.runExpensiveCodeResults
+    }
+
+
+There are some severe limits to this-
+
+*   Each worker has it's own `global` state, and code will bounce between at least four nodes at a time. This means data placed in global one tick will only have a 25% chance of appear next (although it can be added again until it's on all four nodes, bringing the change up to 100%) and that deleted data could easily show up again in the next tick.
+*   The `global` object gets reset fairly regular, meaning all of the data will regularly disappear. The `global` object can not be considered persistent storage.
+*   Although the `global` object is reset often it is only reset on a consistent schedule. Data placed in it may last longer than expected.
+
+These limitions make the `global` object ideal for certain types of caching, such as when the result of a function is always going to be the same or if it doesn't matter when "stale" data is used then storing the data in global. For cases where results may change and data gets invalidated meta data- such as a TTL or version identifier- will have to be stored alongside the results to facilitate that.
+
+
+## Require Cache
+
+Every time `require` is called the results are cached. This reduces the load on the server and your script as it prevents the various javascript modules from having to be compiled every tick. The require cache itself is tied to the `global` cache and gets cleared at the same time, although there are some circumstances where the `require` cache will get cleared (fully or partially) while the `global` cache does not.
+
+From a performance standpoint the fact that the `require` and `global` caches clear at once means that `global resets` are particularly expensive events.
+
+
+## Tips
+
+*   Be extremely careful what you cache in Memory, as the Memory Parse time can be expensive.
+*   Objects are more expensive to parse than strings. Converting items like [RoomPositions](/api/#RoomPosition) to a flat string before caching and then converting back as needed can have a surprisingly large impact.
+*   For extremely large objects with repetitive data- such as [CostMatrixes](api/#PathFinder-CostMatrix)- compression can save a lot of space. Players going this route should look into [lzstring](http://pieroxy.net/blog/pages/lz-string/index.html), and should also make sure they utilize the `global` cache to minimize the amount of times the same costmatrix has to be decompressed.
```

Comment:
As mentioned above, consider having a space before and after the dash, the same goes for the following line.